client/src/pages/Home.jsx

```diff
@@ -5,47 +5,68 @@ import { Icon } from "leaflet";
 import LeafletGeocoder from "../components/LeafletGeocoder";
 import "leaflet/dist/leaflet.css";
 import "../styles/Home.css";
+import ArtDetails from "../components/ArtDetails";
 
 function Home() {
   const [position, setPosition] = useState([
     44.831271602173324, -0.5722962522737938,
   ]);
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedArt, setSelectedArt] = useState(null);
 
   const artData = useLoaderData();
 
-  const artUrl = import.meta.env.VITE_API_URL;
-
   const artIcon = (url) =>
     new Icon({
       iconUrl: url,
       iconSize: [38, 38],
     });
 
+  const artUrl = import.meta.env.VITE_API_URL;
+
   useEffect(() => {
     navigator.geolocation.getCurrentPosition((geoPosition) => {
       const { latitude, longitude } = geoPosition.coords;
       setPosition([latitude, longitude]);
     });
   }, []);
 
+  const openModal = (art) => {
+    setIsOpen(true);
+    setSelectedArt({
+      ...art,
+      image: `${artUrl}${art.image}`,
+    });
+  };
+
+  const closeModal = () => {
+    setIsOpen(false);
+  };
+
   return (
-    <MapContainer center={position} zoom={13} className="leaflet-container">
-      <TileLayer
-        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
-        attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
-      />
-      <Marker position={position}>
-        <Popup>Place de la Victoire</Popup>
-      </Marker>
+    <>
+      <MapContainer center={position} zoom={13} className="leaflet-container">
+        <TileLayer
+          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+          attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+        />
+        <Marker position={position}>
+          <Popup>Vous êtes ici !</Popup>
+        </Marker>
         {artData.map((art) => (
           <Marker
             key={art.id}
             position={[art.latitude, art.longitude]}
             icon={artIcon(`${artUrl}${art.image}`)}
+            eventHandlers={{ click: () => openModal(art) }}
```

Comment:
Didn't know we could use event handlers like that. 
I knew about passing a function and calling a function. But written like that. A first time.
Thanks